### ISSUE TITLE
Fix bug in per-task sparsification training

### ISSUE THREAD

**Issue by jacobrenn** [Beyond-ML-Labs/mann#7]
Currently, function does not prune correctly when asked to prune subsequent tasks

**Comment by jacobrenn** [comment]
It appears this bug is only present when `MultiMaskedConv2D` layers are used.  Gradients are correctly identified when task 0 is selected, but for any later task, gradients are incorrectly calculated within TensorFlow
